### PR TITLE
Update Kubernetes Config Provider to 1.2.0

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR updates the Kubernetes Config Provider bundled in our images to 1.2.0.

### Checklist

- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: docker-images/artifacts/kafka-thirdparty-libs/3.7.0/ignorelist [hunk 1]
```diff
@@ -1,2 +1,2 @@
 jmx_prometheus_javaagent-1.0.1.jar
-snakeyaml-2.0.jar
\ No newline at end of file
+snakeyaml-2.2.jar
```
Comment (ppatierno) [inline docker-images/artifacts/kafka-thirdparty-libs/3.7.0/ignorelist:2]: is this update related to the config provider? Even if not, I am fine having it here.

**Review by ppatierno — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

File: docker-images/artifacts/kafka-thirdparty-libs/3.7.0/ignorelist (hunk 1 above)
Comment (scholzj) [reply docker-images/artifacts/kafka-thirdparty-libs/3.7.0/ignorelist:2]: Yes. The config provider now doesn't depend on the snakeyaml 2.0 but 2.2 library. So I had to update the ignore list as well.

**Review by im-konge — APPROVED** [review]
